crates/bevy_scene/src/bundle.rs

```diff
@@ -0,0 +1,66 @@
+use bevy_asset::Handle;
+use bevy_ecs::{
+    bundle::Bundle,
+    change_detection::ResMut,
+    entity::Entity,
+    prelude::{Changed, Component},
+    system::Query,
+};
+use bevy_transform::components::{GlobalTransform, Transform};
+
+use crate::{DynamicScene, InstanceId, Scene, SceneSpawner};
+
+#[derive(Default, Component)]
+pub struct HasSceneInstance(Option<InstanceId>);
```

Comment:
We can do this either by having a wrapper component around an `HasSceneInstance(Option<InstanceId>)` or have a `SceneInstance(InstanceId)` component that is not in the bundle, but gets added when the scene is spawned.

The latter would have the advantage that it is easier to query for the instance id (just `Query<&SceneInstance>`).

Reply:
I like this strategy more: it's also a lot simpler to understand and could be private.

Reply:
having the component on the bundle allows for easier discoverability that it exists, but at the cost of it being an option. I think the improved discoverability is worth it

Reply:
I think in this case being able to easily query for "ready scenes" is the better UX. Otherwise we're putting the burden on the user to filter through all "unready scenes". Discoverability is an issue, but thats also true for a number of marker components, and I think we've largely decided that its a ["documentation and editor" issue](https://github.com/bevyengine/bevy/issues/3833)?

Reply:
I agree with Cart's position here.

Reply:
move to a not option 👍